server/src/data_sources/redis_provider.rs

```diff
@@ -50,14 +56,21 @@ impl FeatureSink for RedisProvider {
         features: ClientFeatures,
     ) -> EdgeResult<()> {
         let mut lock = self.redis_client.write().await;
-        let serialized_features = serde_json::to_string(&features)?;
-        let _: () = lock.set(build_features_key(&token.token), serialized_features)?;
+        let mut con = lock.get_async_connection().await?;
+
+        let key = build_features_key(token);
+
+        let features_to_store =
+            if let Some(stored_features) = con.get::<&str, Option<String>>(key.as_str()).await? {
+                let stored_features = serde_json::from_str::<ClientFeatures>(&stored_features)?;
+                stored_features.merge(features.clone())
```

Comment:
Maybe a problem for later, but this could be an always-growing collection. We won't remove data and the TTL might get updated so stored_features will never expire...

Reply:
That's a good point! Yes, we should fix this but probably in a different PR